apis/refs/v1alpha1/organizationref.go

```diff
@@ -27,6 +39,61 @@ type Organization struct {
 	OrganizationName string
 }
 
-func (s *Organization) String() string {
+func (s *Organization) FullyQualifiedName() string {
 	return "organizations/" + s.OrganizationName
 }
+func ResolveOrganization(ctx context.Context, reader client.Reader, src client.Object, ref *OrganizationRef) (*Organization, error) {
+	if ref == nil {
+		return nil, nil
+	}
+
+	if ref.External != "" {
+		if ref.Name != "" {
+			return nil, fmt.Errorf("cannot specify both name and external on organization reference")
+		}
+
+		tokens := strings.Split(ref.External, "/")
+		if len(tokens) == 1 {
+			return &Organization{OrganizationName: tokens[0]}, nil
+		}
+		if len(tokens) == 2 && tokens[0] == "organizations" {
+			return &Organization{OrganizationName: tokens[1]}, nil
+		}
+		return nil, fmt.Errorf("format of organization external=%q was not known (use organization/<orgId> or <orgId>)", ref.External)
+
+	}
+
+	if ref.Name == "" {
+		return nil, fmt.Errorf("must specify either name or external on organization reference")
+	}
+
+	key := types.NamespacedName{
+		Namespace: ref.Namespace,
+		Name:      ref.Name,
+	}
+	if key.Namespace == "" {
+		key.Namespace = src.GetNamespace()
+	}
+
+	organization := &unstructured.Unstructured{}
+	organization.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apigee.cnrm.cloud.google.com",
+		Version: "v1alpha1",
+		Kind:    "ApigeeOrganization",
+	})
+
+	if err := reader.Get(ctx, key, organization); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("referenced Organization %v not found", key)
```

Comment:
Here, we should return a `k8s.NewReferenceNotFoundError`, so that we will immediately requeue once the reference is found. Example: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/26f6f8a355f2a23933e63f48610f915c1022dddd/apis/workstations/v1beta1/cluster_reference.go#L70C15-L70C44

Reply:
Done